### PR TITLE
Update limits basing on Tezos updates [TE-422, TE-405]

### PR DESCRIPTION
**Pull request by akoptelov**
- Limits are updated according to Tezos code
- Encoded data generator is added to test limits
- Now there is the `PeerMessageResponse.message` that contains exactly one specific message

@bkontur Please take a look at the `HISTORY_MAX_SIZE`. It is `200` despite that `CURRENT_BRANCH_HISTORY_MAX_LENGTH` is now `1000`. Should that also be updated?

This addresses following Jiras:
TE-422, TE-405, TE-407

### CODE REVIEW

File: tezos/encoding/src/binary_reader.rs [hunk 1]
```diff
@@ -251,7 +251,7 @@ impl BinaryReader {
                 let mut buf_slice = safe!(buf, upper, buf.take(upper));
                 let res = self.decode_value(&mut buf_slice, inner_encoding);
                 match res {
-                    Err(BinaryReaderError::Underflow { bytes: _ }) => {
+                    Err(BinaryReaderError::Underflow { bytes: _ }) if upper == *max => {
```
Comment (tizoc) [inline tezos/encoding/src/binary_reader.rs:254]: It is weird seeing things go from an underflow error to a "boundary exceeded" one. Reasoning about what the code is doing here makes it clear, but a comment explaining that in this case the inner underflow is caused by the outer-imposed limit would make it easier to understand.

Comment (tizoc) [reply tezos/encoding/src/binary_reader.rs:257]: @akoptelov looks good, thanks

File: tezos/messages/src/p2p/encoding/current_branch.rs [hunk 2]
```diff
@@ -16,7 +16,7 @@ use crate::p2p::encoding::block_header::BlockHeader;
 
 use super::limits::CURRENT_BRANCH_HISTORY_MAX_LENGTH;
 
-pub const HISTORY_MAX_SIZE: u8 = CURRENT_BRANCH_HISTORY_MAX_LENGTH as u8; // 200
+pub const HISTORY_MAX_SIZE: u8 = 200;
```
Comment (tizoc) [inline tezos/messages/src/p2p/encoding/current_branch.rs:19]: if this is supposed to match `CURRENT_BRANCH_HISTORY_MAX_LENGTH`, can we add a static assert to verify that it remains true all the time?

I use this crate in ocaml-interop for that https://docs.rs/static_assertions/1.1.0/static_assertions/index.html

Comment (tizoc) [reply tezos/messages/src/p2p/encoding/current_branch.rs:19]: Oh wait, I just saw the new value for `CURRENT_BRANCH_HISTORY_MAX_LENGTH`, ignore this comment.

File: tezos/messages/src/p2p/encoding/peer.rs [hunk 3]
```diff
@@ -47,74 +49,80 @@ pub struct PeerMessageResponse {
...
-                Tag::new(
-                    0x60,
-                    "GetOperationsForBlocks",
-                    GetOperationsForBlocksMessage::encoding().clone(),
-                ),
-                Tag::new(
-                    0x61,
-                    "OperationsForBlocks",
-                    OperationsForBlocksMessage::encoding().clone(),
+    Encoding::bounded(
```
Comment (tizoc) [inline tezos/messages/src/p2p/encoding/peer.rs:52]: @akoptelov this is the thing we discussed some time ago about a mismatch between our implementation representing this as a list while in OCaml it was a single dynamically-sized element, is that correct?

Comment (tizoc) [reply tezos/messages/src/p2p/encoding/peer.rs:52]: What I am thinking is, why does this need to be a list at all? (I am not that familiar with our encoding library, so there may be details that I am missing)

Comment (akoptelov) [reply tezos/messages/src/p2p/encoding/peer.rs:52]: Well, just limiting the list a very fast fix, with minimal changes. But I agree, replacing the list with single value is more clear and firm solution.

**Comment by tizoc** [comment]
@akoptelov last change looks good, but something happened with the drone run and it failed.

**Comment by tizoc** [comment]
Looks good to me.

@bkontur @akoptelov I'm thinking that I may be worth it to have a small OCaml program that just prints the values of all limits to easily detect when something changes. Probably as part of the idea I mentioned some time ago about using OCaml's PPX to produce parseable definitions of the encodings. I'm still dealing with storage integration related stuff, but it is something I can look into doing once I'm free from other things.

File: tezos/messages/src/p2p/encoding/current_branch.rs (hunk 2 above)
Comment (bkontur) [inline tezos/messages/src/p2p/encoding/current_branch.rs:19]: @akoptelov 
could we remove HISTORY_MAX_SIZE now?  CURRENT_BRANCH_HISTORY_MAX_LENGTH is what we want

**Review by bkontur — CHANGES_REQUESTED** [review]
@akoptelov 
please to develop,
and check comment about  HISTORY_MAX_SIZE

File: tezos/messages/src/p2p/encoding/current_branch.rs (hunk 2 above)
Comment (akoptelov) [reply tezos/messages/src/p2p/encoding/current_branch.rs:19]: That constant is still used in Tezos code:
```
(* FIXME: this should not be hard-coded *)
let max_locator_size = 200
```

Comment (bkontur) [reply tezos/messages/src/p2p/encoding/current_branch.rs:19]: @akoptelov 
I know, but we have now:
```
pub const HISTORY_MAX_SIZE: u8 = 200;
```
and 
```
pub const CURRENT_BRANCH_HISTORY_MAX_LENGTH: usize = 200;
```

it is the same, so I just want to remove HISTORY_MAX_SIZE and use CURRENT_BRANCH_HISTORY_MAX_LENGTH instead

Comment (akoptelov) [reply tezos/messages/src/p2p/encoding/current_branch.rs:19]: Not exactly. The `CURRENT_BRANCH_HISTORY_MAX_LENGTH` is actually `1000`.

**Review by bkontur — APPROVED** [review]
@akoptelov 
wait for CI and please rebase and merge to develop

**Pull request merged**